service/lib/dinstaller/software.rb

```diff
@@ -169,5 +183,26 @@ def find_products
       logger.info "Supported products found: #{supported_products.map(&:name).join(",")}"
       supported_products
     end
+
+    REPOS_BACKUP = "/etc/zypp/repos.d.dinstaller.backup"
+    private_constant :REPOS_BACKUP
+
+    REPOS_DIR = "/etc/zypp/repos.d"
+    private_constant :REPOS_DIR
+
+    # ensure that repos backup is there and repos.d is empty
+    def store_original_repos
+      # Backup was already created, so just remove all repos
+      if File.directory?(REPOS_BACKUP)
```

Comment:
Maybe we should log the store/restore actions. I kind of expect that to become relevant for debugging at some point.

Reply:
makes sense for me.